packages/office-ui-fabric-react/src/components/DetailsList/DetailsList.types.ts

```diff
@@ -501,3 +506,15 @@ export enum CheckboxVisibility {
    */
   hidden = 2
 }
+
+export type IDetailsListStyleProps = Required<Pick<IDetailsListProps, 'theme'>> &
+  Pick<IDetailsListProps, 'className'> & {
+    isHorizontalConstrained?: boolean;
```

Comment:
We really should be documenting these as well, if you think about the purpose these props serve and from the POV of someone writing a style function for DetailsList

Reply:
added more docs to these